doc/release/upcoming_changes/25193.compatibility.rst

```diff
@@ -0,0 +1,4 @@
+The ``f2py.compile()`` helper has been removed due to a severe restriction in
+functionality after the ``meson`` switch. Users are urged to replace these calls
+with ``subprocess.run`` calls to ``python -m numpy.f2py`` instead with
+environment variables set to interact with ``meson``.
```

Comment:
Hmm. The use of environment variables is problematic. Isn't there any other way to hook into meson?

Reply:
There are [native files](https://mesonbuild.com/Machine-files.html), I've added a note.